### ISSUE TITLE
Use `sourceTensor.clone().detach()` to construct tensors

### ISSUE THREAD

**Issue by TianyuDu** [Lightning-AI/lightning#14704]
## Proposed refactor

I encountered the PyTorch warning 
```
/home/tianyudu/anaconda3/envs/production/lib/python3.10/site-packages/pytorch_lightning/core/module.py:555: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  value = torch.tensor(value, device=self.device)
```
after upgrading to lightning `1.7.5` and PyTorch version `1.12.1`. The fix should be pretty easy: replacing the `value = torch.tensor(value, device=self.device)` at `pytorch_lightning/core/module.py:555` with the suggested change proposed by PyTorch.

The package is still usable after suppressing the warning : )
______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/Lightning-AI/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/Lightning-AI/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/Lightning-AI/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/Lightning-AI/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging PyTorch Lightning, Transformers, and Hydra.

**Comment by carmocca** [comment]
This is fixed in the [1.7.6 release](https://github.com/Lightning-AI/lightning/releases/tag/1.7.6). Thank you for reporting!
